modin/pandas/utils.py

```diff
@@ -430,5 +430,31 @@ def _doc_binary_op(operation, bin_op, left="Series", right="right", returns="Ser
     return doc_op
 
 
+def apply_function_on_selected_items(input_list, filter_condition, func_apply):
+    """
+    Apply a funtion the provided list for elements filtered by a filter condition.
+
+    Parameters
+    ----------
+    input_list : list
+        The list that has to be manipulated.
+    filter_condition : callable
+        The filter function to filter elements in the given list.
+    func_apply : callable
+        The function that has to be applied to elements in the list.
+    """
+    if not input_list:
+        return
+    filtered_list = []
+    filtered_idx = []
+    for idx, item in enumerate(input_list):
+        if filter_condition(item):
+            filtered_idx.append(idx)
+            filtered_list.append(item)
+    filtered_list = func_apply(filtered_list)
+    for idx, item in zip(filtered_idx, filtered_list):
+        input_list[idx] = item
```

Comment:
Why make changes in place? If there is no strong reason, then let's return it through `return` statement.

Reply:
Did inplace with the idea to not create an additional list and to avoid a copy, so the space complexity is reduced, I have now changed code to return the modified list from the function.

Reply:
You still allocate memory for variables `filtered_idx` and `filtered_list`. In this case, the memory savings are not so significant.

You can simply rewrite this function with list comprehension.

```python
input_list = [func_apply(item) if filter_condition(item) else item for item in input_list]
return input_list
```

Reply:
Ah I see, sorry.

However, you should not return something that is implicitly changed; this option is even more unclear than the previous option. The best option is not to change the input data and return the result explicitly. I still think that avoiding copying the input list is premature optimization.